src/main/java/frc/robot/commands/CmdManipulatorSetIntakeSpeed.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Manipulator;
+
+public class CmdManipulatorSetIntakeSpeed extends CommandBase 
+{
+    private final Manipulator _manipulatorSubsystem;
+    private final double      _intakeSpeed;
+
+    public CmdManipulatorSetIntakeSpeed(double intakeSpeed) 
+    {
+        _manipulatorSubsystem = Manipulator.getInstance();
+        _intakeSpeed          = intakeSpeed;
+    }
+
+    @Override
+    public void initialize() 
+    {
+        _manipulatorSubsystem.setIntakeSpeed(_intakeSpeed);
+    }
+
+    @Override
+    public void end(boolean interrupted) 
+    {
+        if(_manipulatorSubsystem.isIntakeSensorActive())
```

Comment:
This check should happen in the `isFinished` function. You'll still want to set the intake speed to 0 here.